base/common/python/pki/nssdb.py

```diff
@@ -440,37 +444,59 @@ def convert_db(self):
 
         logger.info("Migration successful")
 
-    def module_exists(self, name):
+    def module_exists(self, name, library=None):
 
-        logger.info('Checking module %s', name)
+        def regex_in_output(cmd, pattern):
+            result = self.run(cmd, capture_output=True, check=True)
+            output = result.stdout.decode('utf-8')
 
-        cmd = [
-            'modutil',
-            '-dbdir', self.directory,
-            '-rawlist'
-        ]
+            p = re.compile(pattern)
+            logger.info('Looking for: %s', pattern)
 
-        result = self.run(cmd, capture_output=True, check=True)
-        output = result.stdout.decode('utf-8')
+            for line in output.splitlines():
+                line = line.strip()
 
-        pattern = re.compile(r' name="%s"' % name)
+                if not line:
+                    continue
 
-        for line in output.splitlines():
-            line = line.strip()
+                logger.info('Output: %s', line)
 
-            if not line:
-                continue
+                if p.search(line):
+                    return True
 
-            logger.info('Output: %s', line)
+            return False
 
-            if pattern.search(line):
+        logger.info('Checking module %s', name)
+
+        commands = [
+            [
+                'modutil',
+                '-dbdir', self.directory,
+                '-rawlist'
+            ],
+            [
+                'p11-kit',
+                'list-modules'
+            ],
+        ]
+
+        patterns = [
+            r'name="%s"' % name,
+            r'%s: %s' % (name, library if library else '')
```

Comment:
Does the `library` value need to be escaped since it will be used as a regex pattern?

Reply:
That's a fair point. Do you think I should re.escape() the name as well while I'm at it? I couldn't find any restriction on the characters used in a token name.

Reply:
Yeah, if it works feel free to escape the `name` too. I think the `library` is more risky since it can contain `/` which is a special character for regex.

Reply:
I tested and at least in my simple case it works fine. I don't know if someone would, for example, use paranthesis in a token name which is also a special character.

Reply:
Please just escape both values, and enjoy better sleep :)

Reply:
They are currently escaped when the regex is compiled.

Reply:
Thank you.